### PR TITLE
Model registry concept

### PR DESCRIPTION
**Pull request by dpanshug**
Closes: [RHOAIENG-2116](https://issues.redhat.com/browse/RHOAIENG-2116)

Collaborated with @manaswinidas and @lucferbux 

## Description
Created concept page Model Registry with context and apiHooks

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by dpanshug** [comment]
This PR is still WIP, need to merge #2599 first

File: frontend/src/concepts/modelRegistry/context/useModelRegistryNamespaceCR.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const resourceLoaded = state[1] && !!state[0];
+  const hasStatus = isModelRegistryAvailable(state);
+  React.useEffect(() => {
+    setIsStarting(resourceLoaded && !hasStatus);
+  }, [hasStatus, resourceLoaded]);
+
+  return state;
+};
+
+export const hasServerTimedOut = (
```
Comment (lucferbux) [inline frontend/src/concepts/modelRegistry/context/useModelRegistryNamespaceCR.ts:56]: We probably don't need this, this is a workaround for pipelines, hopefully it won't be needed for us, we can safely remove this in teh future

**Review by lucferbux — APPROVED** [review]
/lgtm

File: frontend/src/concepts/modelRegistry/context/useModelRegistryNamespaceCR.ts (hunk 1 above)
Comment (lucferbux) [reply frontend/src/concepts/modelRegistry/context/useModelRegistryNamespaceCR.ts:56]: @dpanshug FYI, you can resolve the conversation but keep this noted so we can remove it later.

**Pull request merged**